lib/google_search_data_viewer/keyword/search_results.ex

```diff
@@ -7,4 +7,20 @@ defmodule GoogleSearchDataViewer.Keyword.SearchResults do
     |> SearchResultUrl.changeset()
     |> Repo.insert()
   end
+
+  def group_by_adword_types(search_result_urls) do
+    search_result_urls
+    |> Enum.group_by(&{&1.is_adword, &1.is_top_adword})
+    |> Enum.map(&add_adword_type_from_group/1)
+    |> Map.new()
```

Comment:
Not sure, will this work? so we can remove the extra two functions.
```elixir
    def group_by_adword_types(search_result_urls) do
        Enum.group_by(search_result_urls, &group_adword_type/1)
    end

    defp group_adword_type(%SearchResultUrl{is_adword: true, is_top_adword: true}), do: :top_adwords
    defp group_adword_type(%SearchResultUrl{is_adword: false, is_top_adword: false}), do: :non_adwords
    defp group_adword_type(%SearchResultUrl{is_adword: true, is_top_adword: false}), do: :adwords
```

Reply:
Yes, a lot better!